cmd/clusterctl/client/cluster/template.go

```diff
@@ -181,35 +182,77 @@ func (t *templateClient) getGitHubFileContent(rURL *url.URL) ([]byte, error) {
 
 	// Extract all the info from url split.
 	owner := urlSplit[0]
-	repository := urlSplit[1]
+	repo := urlSplit[1]
+	linkType := urlSplit[2]
 	branch := urlSplit[3]
 	path := strings.Join(urlSplit[4:], "/")
 
 	// gets the GitHub client
-	client, err := t.gitHubClientFactory(t.configClient.Variables())
+	ghClient, err := t.gitHubClientFactory(t.configClient.Variables())
 	if err != nil {
 		return nil, err
 	}
 
 	// gets the file from GiHub
-	fileContent, _, _, err := client.Repositories.GetContents(context.TODO(), owner, repository, path, &github.RepositoryContentGetOptions{Ref: branch})
+	switch linkType {
+	case "blob": // get file from a code in a github repo
+		return getGithubFileContentFromCode(ghClient, rURL.Path, owner, repo, path, branch)
+
+	case "releases": // get a github release asset
+		tag := urlSplit[4]
+		assetName := urlSplit[5]
+
+		return getGithubAssetFromRelease(ghClient, rURL.Path, owner, repo, tag, assetName)
+
+	default:
+		return nil, fmt.Errorf("unknown github URL: %v", rURL)
+	}
+}
+
+func getGithubFileContentFromCode(ghClient *github.Client, fullPath string, owner string, repo string, path string, branch string) ([]byte, error) {
+	fileContent, _, _, err := ghClient.Repositories.GetContents(ctx, owner, repo, path, &github.RepositoryContentGetOptions{Ref: branch})
 	if err != nil {
-		return nil, handleGithubErr(err, "failed to get %q", rURL.Path)
+		return nil, handleGithubErr(err, "failed to get %q", fullPath)
 	}
 	if fileContent == nil {
-		return nil, errors.Errorf("%q does not return a valid file content", rURL.Path)
+		return nil, errors.Errorf("%q does not return a valid file content", fullPath)
 	}
 	if fileContent.Encoding == nil || *fileContent.Encoding != "base64" {
-		return nil, errors.Errorf("invalid encoding detected for %q. Only base64 encoding supported", rURL.Path)
+		return nil, errors.Errorf("invalid encoding detected for %q. Only base64 encoding supported", fullPath)
 	}
-
 	content, err := base64.StdEncoding.DecodeString(*fileContent.Content)
 	if err != nil {
-		return nil, errors.Wrapf(err, "failed to decode file %q", rURL.Path)
+		return nil, errors.Wrapf(err, "failed to decode file %q", fullPath)
 	}
 	return content, nil
 }
 
+func getGithubAssetFromRelease(ghClient *github.Client, path string, owner string, repo string, tag string, assetName string) ([]byte, error) {
+	release, _, err := ghClient.Repositories.GetReleaseByTag(ctx, owner, repo, tag)
```

Comment:
should we check for a non-nil `release`?

Reply:
done